app/src/utils/services/DataService.ts

```diff
@@ -46,11 +46,16 @@ class DataService {
                                 location: Location|undefined,
                                 radius: string
                             }): Promise<any> {
-        const query = new QueryBuilder();
-        searchValues.forEach((value) => {
-            query.shouldMatch('categories', value)
-                .shouldMatch('title', value);
-        });
+        const query: QueryBuilder = new QueryBuilder();
+        console.log(searchValues);
+        if (searchValues.join(',') !== '') {
+            searchValues.forEach((value) => {
+                query.shouldMatch('categories', value)
+                    .shouldMatch('title', value);
+            });
+          } else {
+            this.performQuery(query);
```

Comment:
Missing return statement?

Reply:
I would not add an return statement there because this would prevent the user to search for all posts within a region (f.e. when searching for nothing but specifying location and range). If we return there the user would consistently get all posts despite him filtering by location and range

Reply:
okay so why then is the query being executed at this point?

Reply:
I was under the impression that this line queries all result into the query variable. Then queryObject utilizes these results and does filtering based on location and range. So its, in the worst case, two queries which is admittedly is not optimal performance wise.

Reply:
The line is just straignt useless. Kicking it out